### DISCUSSION TITLE
Redirect on error=login_required

### DISCUSSION THREAD

**Discussion by hf409** [OpenIDC/mod_auth_openidc#625] [Q&A] [answered]
Hi all,

I need to do some further processing when setting prompt=none and the return is oidcRedirect?error=login_required.
Although I set OIDCDefaultURL, it does not redirect, but shows the error page.
From reading the code, I guess redirecting to OIDCDefaultURL is not supposed to work for this situation.
Is there a way to do it anyway?

Thanks for any help,
Heike

**Comment by zandbelt** [comment, marked as answer]
using a LocationMatch on the query parameter value should work so you can have Apache redirect to a different page

**Reply by hf409** [reply]
I tried that and ended in something like
```
<LocationMatch ^/oidcRedirect/ >
    <If "%{QUERY_STRING} =~ /error=login_required/">
        RewriteRule (.*)  https://example.com/mySpecialErrorPage  [R=302,L]
    </If>
    <Else>
        AuthType openid-connect
        Require valid-user
    </Else>
</LocationMatch>
```
It seems a bit strange to have the auth config within if/else. But it works. I don't use it for real authentication, just
to test if it succeeds. So, it's ok for me.

Thanks for your help :-)
Heike
